### PR TITLE
Fixed issues in ziputils class and added proper exceptions.

### PR DESCRIPTION
**Pull request by rnavagamuwa**
Custom exceptions added for APIClientGeneration class. Fixed some issues in ziputils class.

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIClientGenerationException.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package org.wso2.carbon.apimgt.impl;
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIClientGenerationException.java]: Please add licence header

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIClientGenerationManager.java [hunk 2]
```diff
@@ -51,7 +52,8 @@
 public class APIClientGenerationManager {
     private static final Log log = LogFactory.getLog(APIClientGenerationManager.class);
 
-    public String sdkGeneration(String appName, String sdkLanguage, String userName, String groupId) {
+    public String sdkGeneration(String appName, String sdkLanguage, String userName, String groupId)
+            throws APIClientGenerationException, IOException {
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIClientGenerationManager.java]: Why we throtting IO exception?

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIClientGenerationManager.java [hunk 3]
```diff
@@ -177,16 +186,21 @@ public String sdkGeneration(String appName, String sdkLanguage, String userName,
...
                 }
                 fileWriter = new FileWriter(spec.getAbsoluteFile());
-                BufferedWriter bufferedWriter = new BufferedWriter(fileWriter);
+                bufferedWriter = new BufferedWriter(fileWriter);
                 bufferedWriter.write(swagger);
-                bufferedWriter.close();
             } catch (IOException e) {
                 log.error("problem when storing the temporary swagger file", e);
+                throw new APIClientGenerationException("problem when storing the temporary swagger file", e);
+            }finally {
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIClientGenerationManager.java]: formating issue

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIClientGenerationManager.java [hunk 4]
```diff
@@ -51,7 +52,8 @@
 public class APIClientGenerationManager {
     private static final Log log = LogFactory.getLog(APIClientGenerationManager.class);
 
-    public String sdkGeneration(String appName, String sdkLanguage, String userName, String groupId) {
+    public String sdkGeneration(String appName, String sdkLanguage, String userName, String groupId)
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIClientGenerationManager.java]: add method coomments

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIClientGenerationManager.java (hunk 3 above)
Comment (rnavagamuwa) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIClientGenerationManager.java]: Fixed

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIClientGenerationManager.java (hunk 4 above)
Comment (rnavagamuwa) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIClientGenerationManager.java]: Added

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIClientGenerationManager.java (hunk 2 above)
Comment (rnavagamuwa) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIClientGenerationManager.java]: I have removed and used my custom exception class instead

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIClientGenerationException.java (hunk 1 above)
Comment (rnavagamuwa) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIClientGenerationException.java]: Added

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIClientGenerationManager.java [hunk 5]
```diff
@@ -177,16 +196,26 @@ public String sdkGeneration(String appName, String sdkLanguage, String userName,
...
             } catch (IOException e) {
                 log.error("problem when storing the temporary swagger file", e);
+                throw new APIClientGenerationException("problem when storing the temporary swagger file", e);
+            } finally {
+                if (bufferedWriter != null) {
+                    try {
+                        bufferedWriter.close();
+                    } catch (IOException e) {
+                        log.error("problem when closing the connection which is opened to store the swagger file", e);
+                        throw new APIClientGenerationException("problem when closing the connection which is opened to store the swagger file", e);
```
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIClientGenerationManager.java]: We don't have to throw an exception inside the finally block here. Please remove it.

Comment (rnavagamuwa) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIClientGenerationManager.java]: Fixed

**Pull request merged**